### PR TITLE
[IOD-103] Google Login and Edition Permissions

### PR DESCRIPTION
**Pull request by Szyntos**
Added rudimentary logging system using firebase. To every request to the server should be attached a token from LocalStorage. Somewhat tested. 

Added Filtering by user only to EditionListCreateAPIView, more coming soon.

Permissions are done with UserEdition table where 

User has Edition with Permission

User1 has Edition2 with permission.VIEW

Admin is done by

UserAdmin has None with permission.ADMIN

in def google_login(request) there is a sleep(2), because of a bug in FireBase api https://stackoverflow.com/questions/71915309/token-used-too-early-error-thrown-by-firebase-admin-auths-verify-id-token-metho

https://github.com/Jullija/ConspectsManager/assets/40152296/4d9de618-d9d0-42f3-a0ca-3e28d77c7783

### CODE REVIEW

File: backend/conspects/serializers.py
Comment (dadamczykk) [inline backend/conspects/serializers.py:111]: There are three similar definitions of this method? couldn't it be extracted into another helper function and ```get_user_permission``` be a wrapper?

File: backend/conspects/views.py
Comment (dadamczykk) [inline backend/conspects/views.py:149]: Please decide if those comments are necessary. If you want to merge as final verion and those commens are unnecessary, remove them.

File: backend/users/apps.py
Comment (dadamczykk) [inline backend/users/apps.py:13]: Make sure there are empty line at the end of files

File: backend/users/models/user_edition.py
Comment (dadamczykk) [inline backend/users/models/user_edition.py:15]: I'm not sure if those should be ```on_delete=models.CASCADE```, wouldn't it delete edition and user if we take away permission from user with view/edit access?

**Comment by dadamczykk** [comment]
Lets wait for @AnCichocka frontend review and @olgierdjw backend before merging

File: backend/conspects/views.py
Comment (Szyntos) [reply backend/conspects/views.py:149]: nope, they are not, i forgor 💀

File: backend/conspects/serializers.py
Comment (Szyntos) [reply backend/conspects/serializers.py:111]: Fair point, it could be extracted

File: backend/users/models/user_edition.py
Comment (Szyntos) [reply backend/users/models/user_edition.py:15]: I tested removing permissions, but will test further and change if neccesary

**Review by AnCichocka — COMMENTED** [review]
It's a little pity that this PR wasn't separated to smaller ones.

I don't really understand how this share function works:
image

+ the concept was a bit different?
image

Over all great work. Introducing axios client, user id provider and types is worth considering. I believe it would be best to wark through this PR on stand up meeting?

**Review by Jullija — APPROVED** [review]
super duper
image

File: frontend/src/screens/AccessRightsPanel/AccessRightsPanel.tsx
Comment (Szyntos) [reply frontend/src/screens/AccessRightsPanel/AccessRightsPanel.tsx:120]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Object_initializer#computed_property_names

File: backend/users/models/user_edition.py
Comment (Szyntos) [reply backend/users/models/user_edition.py:15]: The on_delete=models.CASCADE argument in both ForeignKey fields (user and edition) does indeed specify that if a referenced User or Edition object is deleted, then the related UserEdition instances should also be deleted. However, it's important to clarify that this does not work the other way around: deleting a UserEdition instance does not trigger a deletion of the associated User or Edition. ~ Chat (i know that it's not the best source of info, but it makes sense)

File: frontend/src/screens/AccessRightsPanel/AccessRightsPanel.tsx
Comment (dadamczykk) [reply frontend/src/screens/AccessRightsPanel/AccessRightsPanel.tsx:120]: @Szyntos Quite interesting, I haven't heard about computed properties earlier

Comment (Szyntos) [reply frontend/src/screens/AccessRightsPanel/AccessRightsPanel.tsx:120]: Me niether 😅

**Comment by AnCichocka** [comment]
You created axiosClient and do not use it instead of axios.get..., can be corrected while resolving conflicts (there is axiosClient on main now). As long as it works it's good with me!

**Pull request merged**
